ConwaysGameOfLife/InputParser.cs

```diff
@@ -3,32 +3,32 @@
 
 namespace ConwaysGameOfLife
 {
-    public class InputParser
+    public static class InputParser
     {
         // parses the input received 
-        public List<Location> ParseLocations(string userInput)
-        {
-            var liveCellLocations = new List<Location>();
-            var locationStringArray = SplitInput(userInput, " ");
-            for (var i = 0; i < locationStringArray.Length; i++)
-            {
-                var location = SplitInput(locationStringArray[i], ",");
-
-                liveCellLocations.Add(new Location(ParseInput(location[0]), ParseInput(location[1])));
-            }
-            return liveCellLocations;
-        }
 
-        public Universe ParseUniverse(string userInput)
+        public static Universe ParseUniverse(string userInput)
         {
             var universe = SplitInput(userInput, ",");
+            var gridWidth = ParseInput(universe[0]);
+            var gridLength = ParseInput(universe[1]);
+            ThrowException(gridWidth, gridLength);

```

Comment:
rename/refactor this bit so it's clearer that it's making sure that x and y are > 0 before throwing an exception. Alternatively it can go into the validation class as it's validating the width and length - but it depends on what you consider as validating vs parsing - so up to you. Would like your thoughts on this too @Kedoodle

Reply:
Have put thoughts in a previous comment in the validator tests. Would like clearer separation of responsibilities between the `Validator` and the `InputParser`. I see the `Validator` working as a pre-check to the `InputParser` - attempting to validate after a "successful" parse can only catch when the format is correct (and an object is actually initialised out of a string), but what about when the format is wrong? At the moment, an exception is thrown when the format is wrong - no object is initialised for the `Validator` to validate. In this case, the `InputParser` implicitly does the validation because it throws an exception when the format is wrong, not giving the `Validator` a chance to do any validation.